bokehjs/src/coffee/models/glyphs/segment.coffee

```diff
@@ -41,13 +41,15 @@ export class SegmentView extends GlyphView
       y: this.renderer.plot_view.canvas.vy_to_sy(vy)
 
     hits = []
-
-    candidates = @index.indices({minX: x, minY: y, maxX: x, maxY: y})
+    lw_voffset = 2 # FIXME: Use maximum of segments line_width/2 instead of magic constant 2
+    lw_sxoffset = @renderer.xscale.invert(vx-lw_voffset, true)
+    lw_syoffset = @renderer.yscale.invert(vy-lw_voffset, true)
+    candidates = @index.indices({minX: lw_sxoffset, minY: lw_syoffset, maxX: 2*x - lw_sxoffset, maxY: 2*y - lw_syoffset})
```

Comment:
Can you explain this a bit more? I am confused why you'd multiply the *coordinate* by 2? If the intent is to check a box around the point `x,y` then I would expect maybe:
```
candidates = @index.indices({
    minX: x - lw_sxoffset, 
    minY: y - lw_syoffset, 
    maxX: x + lw_sxoffset, 
    maxY: x + lw_syoffset})
```

Reply:
Oh I see `lw_sxoffset` already has the displacement baked in. I still think the right way to do things is then to transform all the values independently:
```
candidates = @index.indices({
    minX: @renderer.xscale.invert(vx-lw_voffset, true), 
    minY: @renderer.xscale.invert(vx-lw_voffset, true),
    maxX: @renderer.xscale.invert(vx+lw_voffset, true),
    maxY: @renderer.xscale.invert(vx+lw_voffset, true)
```